homeassistant/components/samsungtv/media_player.py

```diff
@@ -71,13 +71,27 @@
     ):
         turn_on_action = hass.data[DOMAIN][ip_address][CONF_ON_ACTION]
         on_script = Script(hass, turn_on_action)
-    async_add_entities([SamsungTVDevice(config_entry, on_script)])
+
+    # Initialize bridge
+    data = config_entry.data.copy()
+    bridge = SamsungTVBridge.get_bridge(
+        data[CONF_METHOD], data[CONF_HOST], data[CONF_PORT], data.get(CONF_TOKEN),
+    )
+    if bridge.port is None and bridge.default_port is not None:
```

Comment:
Shouldn't we check if the port is missing in the config entry? That should be the single source of thruth.

Reply:
I don't understand your question sorry. 
For legacy devices the port is and was none. Port is now required  only for websocket devices (it was none before my PR). 
I can make here a simple test (set port to 8001 if method is websocket and port none) before creating the Bridge. Is this what you are asking now? 

PS: IMHO my first single line commit was the best solution... Simple yet effective